src/main/webapp/scripts/map.js

```diff
@@ -78,15 +105,18 @@ function addMarker(map) {
   /* Iterate through stored recommendations to get details */
   for (i = 0; i < recommendations.length; i++) {
     const recommendation = recommendations[i];
-    const placeCategory = recommendation[0];
-    const placeName = recommendation[1];
-    const placeLatLng = new google.maps.LatLng(recommendation[2], recommendation[3]);
-    const placeRecommendation = recommendation[4];
+    const placeCategory = recommendation.category;
+    const placeName = recommendation.name;
+    const placeLatLng = new google.maps.LatLng(recommendation.lat, recommendation.lng);
+    const placeRecommendation = recommendation.description;
+    const placeCost = recommendation.costRating;
+    const placeCrowd = recommendation.crowdRating;
```

Comment:
Please remove these unused variables. it's fine to have them inside the recommendation object but there's no need to read them here without an actual usage.